lib/utils/update-notifier.js

```diff
@@ -113,3 +115,11 @@ const updateNotifier = module.exports = async (npm, spec = 'latest') => {
 
   return messagec
 }
+
+// only update the notification timeout if we actually finished checking
+module.exports = async npm => {
+  const notification = await updateNotifier(npm)
+  // intentional.  do not await this.  it's a best-effort update.
+  updateTimeout(npm)
+  return notification
```

Comment:
```suggestion
  npm.updateNotification = notification
```

Reply:
Why would it be preferable for this file to update an object it doesn’t own??

Reply:
The `updateNotifier` method is, effectively, a method that is part of what we do to build the npm object, which has just been abstracted out to a separate file for ease of testing and management.

Its area of concern is just as constrained if it adds a property to the object it's given, or if it returns a promise to the property's value.  In fact, that concern is somewhat _more_ separated and isolated if the use of the method is just "do the thing" rather than "do the thing, await the result, then add it to the object you care about".